torch/fx/experimental/proxy_tensor.py

```diff
@@ -1827,8 +1793,38 @@ def get_torch_dispatch_modes() -> List[TorchDispatchMode]:
     return torch.utils._python_dispatch._get_current_dispatch_mode_stack()
 
 
-def get_innermost_proxy_mode() -> ProxyTorchDispatchMode:
-    return torch._C._get_dispatch_mode(torch._C._TorchDispatchModeKey.PROXY)
+# TODO: this is a legacy name, there is only ever one proxy mode as it's an
+# infra mode
+def get_innermost_proxy_mode() -> Optional[ProxyTorchDispatchMode]:
+    return get_proxy_mode()
+
+
+def get_proxy_mode() -> Optional[ProxyTorchDispatchMode]:
+    """
+    Current the currently active proxy tracing mode, or None if
+    we are not currently tracing.  This includes pre-dispatch proxy
+    tracing.
+    """
+    pre_dispatch_mode = torch._ops._get_dispatch_mode_pre_dispatch(torch._C._TorchDispatchModeKey.PROXY)
```

Comment:
This is important. Only querying `_get_dispatch_mode` returns None when pre-dispatch tracing is in effect. But it's still important to report the proxy tensor mode even if it is pre-dispatch, for the primary use of this (triggering handle_sym_dispatch).

cc @tugsbayasgalan

Reply:
we actually have a util for this today (that does the proper assertion that the mode doesn't live on both stacks) https://github.com/pytorch/pytorch/blob/main/torch/utils/_python_dispatch.py#L134

`_get_infra_mode(torch._C._TorchDispatchModeKey.PROXY)`